crates/sui/src/upgrade_compatibility.rs

```diff
@@ -687,7 +729,13 @@ fn diag_from_error(
             compiled_unit_with_source,
             lookup,
         ),
-        _ => todo!("Implement diag_from_error for {:?}", error),
+
+        UpgradeCompatibilityModeError::FunctionEntryCompatibility { .. } => {
+            unimplemented!("entry functions can be changed during upgrade")
```

Comment:
nit: this should be an `unreachable` instead of an `unimplemented` I think.

Reply:
Was a little confused based on the first comment on this file, figured it'd be better to just write a small `to_diag` for this case then if the underlying rules change to `check_private_entry = true` for whatever reason users won't get hit with an error.